### PR TITLE
Allow JVM reflection to fix JDK 17 changes.

### LINKED ISSUE #676: Debugger issues in Java 17
As noted by @mbeckerle inside of issues:

- https://github.com/apache/daffodil-vscode/issues/648
- https://github.com/apache/daffodil-vscode/issues/649

So things are not working properly when using Java 17. This issue is for looking into why that is.

### PR DESCRIPTION
**Pull request by arosien**
Allow JVM reflection to fix JDK 17 changes.
    
Certain reflection (used by JAXB) isn't allowed by default in JDK 17. See https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B.
    
Fixes #676.
    
The added JVM flags are ignored by earlier JVMs... except JDK 8 which complains. Ugh.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -118,6 +118,10 @@ lazy val xjcSettings =
     xjcCommandLine += "-p",
     xjcCommandLine += "org.apache.daffodil.tdml",
     xjcBindings += "debugger/src/main/resources/bindings.xjb",
+    xjcJvmOpts ++= Seq(
+      // workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17: https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
```
Comment (mbeckerle) [inline build.sbt:122]: Just break the too-long comment lines here and below.

**Review by mbeckerle — APPROVED** [review]
+1

File: build.sbt (hunk 1 above)
Comment (arosien) [reply build.sbt:122]: The URL is very long, but I'll make the rest break better.

**Comment by arosien** [comment]
I suspect CI is failing because it uses JDK 8 and that JVM flag isn't recognized for that version.

**Comment by arosien** [comment]
> I suspect CI is failing because it uses JDK 8 and that JVM flag isn't recognized for that version.

Yes, this is the case. Ugh. The flag is needed at build time, to generate the JAXB sources, and also at runtime, when dealing with XML via JAXB.
- At build time we could detect the current JVM version and conditionally add the flags.
- At runtime we can't know what JVM version the user has, so we can document the error and have an extension setting that adds the flags when set.

This adds more evidence that JAXB needs to be removed, a la #594.

**Comment by scholarsmate** [comment]
If we can rip and replace the code that is causing the problems without loosing features, that would be the easiest and preferred path.

We had discussed the possibility of bundling a JRE, or compiling it native using Graal VM (https://lists.apache.org/thread/nh8t4c32ov0c0hrvrp2qo3q84179ngvv).  The discussion on JRE control was overtaken by events since we came up with a "lighter touch" solution to the problem at that time and we didn't want to hold up the release.

**Review by scholarsmate — APPROVED** [review]

**Review by shanedell — APPROVED** [review]
+1

File: build.sbt [hunk 2]
```diff
@@ -106,6 +106,24 @@ lazy val debugger = project
...
+lazy val isAtLeastJava17: Boolean = javaMajorVersion >= 17
+/* Workaround: certain reflection (used by JAXB) isn't allowed by default in JDK 17:
+ * https://docs.oracle.com/en/java/javase/17/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+ *
+ * While we can handle this JVM quirk at build time, at runtime we won't know
+ * a user's JVM version. We'll provide documentation and an extension setting
+ * to add these flags to the extension-launched debugger backend.
+ */
+lazy val extraXjcJvmOpts: Seq[String] =
+  if (isAtLeastJava17)
```
Comment (stevedlawrence) [inline build.sbt:120]: This can be replaced with `scala.util.Properties.isJavaAtLeast("17")`, which has a lot of extra special case handling.

Comment (arosien) [reply build.sbt:120]: Wow, TIL!

**Pull request merged**